### DISCUSSION TITLE
Inheriting From Multiple Classes

### DISCUSSION THREAD

**Discussion by jordantshaw** [sqlalchemy/sqlalchemy#8472] [Usage Questions] [answered]
Hi All,

I have an interesting scenario that I need help with.  I am essentially looking for a way to inherit from multiple classes.  This is an example of what Im trying to achieve.  

```
    Base
    /   \
ChildA  ChildB
    \   /
   NewTable
```

So ChildA and ChildB tables inherit from the Base table.  Then I would like to create a NewTable that is basically the Base table with a left join to both child tables.  So NewTable would inherit from ChildA and ChildB.  Im not sure inheritance is the answer or not.  But is there anyway to achieve this?

Currently I am using a subquery to create the class, like in the example below. The problem with this is none of the relationships from the Base table are inherited.
https://docs.sqlalchemy.org/en/14/orm/nonstandard_mappings.html

**Comment by zzzeek** [comment]
the mapped inheritance feature doesn't support this so you would need to use it either with relationships (more likely) or less likely mapping your class directly to the JOIN that you want, but only one of the superclasses can be mapped.

**Reply by jordantshaw** [reply]
Do you have an example of how that would work with relationships?

**Reply by zzzeek** [reply]
you would have NewTable -> ChildA -> Base and then a relationship to ChildB -> Base

**Reply by zzzeek** [reply]
doing diamond inheritance with relational tables itself is a very esoteric idea that I have seen in books before and googling right now im not even finding any examples. still looking

**Reply by jordantshaw** [reply]
Agreed. I don’t think the tables were designed well. They should have been modeled differently. But unfortunately I don’t own the schema and have to work with what is there. 

The problem I see with using relationships, is I need the data to all be one table, as if I was doing two left joins. Or as if tableA and tableB were unioned together, and then joined to the base table.

**Reply by zzzeek** [reply]
cant find it :/

**Reply by zzzeek** [reply]
oh i am just trying to find what such a schema even looks like for examples.    

if this is a legacy database just map to a JOIN of whatever tables you need, then you have your "view".

**Reply by jordantshaw** [reply]
Yeah makes sense. Essentially Base table is a person table. TableA is a child table for person type A. And TableB is a child table of Base containing person type B. So if you union A and B together, you would be able to join 1:1 with base table, if that makes sense.

**Reply by jordantshaw** [reply]
@zzzeek  One more question.  I can almost get what I need just from using relationships from Base and the children table.  My question is, when I query the Base table, and load the relationships, is there anyway in the result set to move the sub-object, into the main result set?

For example:
```
{
    column1: value,
    column2: value,
    column3: value,
    tableA: {
        column10: value,
        column12: value,
    }
}
```
and instead do this?
```
{
    column1: value,
    column2: value,
    column3: value,
    column10: value,
    column12: value
}
```

**Reply by CaselIT** [reply, marked as answer]
You can use association_proxy https://docs.sqlalchemy.org/en/14/orm/extensions/associationproxy.html but you need to list every column manually
